### DISCUSSION TITLE
Source code for fedavg (or fedopt or fedprox)

### DISCUSSION THREAD

**Discussion by pygabc1** [NVIDIA/NVFlare#1727] [Q&A] [answered]
Hi, where I can see the code calculating global-model?  [https://github.com/NVIDIA/NVFlare/blob/2.3/examples/advanced/cifar10/pt/learners/cifar10_learner.py#L396-L409](url) looks aggregating local model to get global model, but where is the specific (source) code for averaging local weights like
![image](https://user-images.githubusercontent.com/31337772/236494159-9d1a2912-d6d0-4a90-bc2b-d0453cb2483f.png)
in 
![image](https://user-images.githubusercontent.com/31337772/236494486-9deafbc4-b016-40e8-8298-cb17be225783.png)
in [https://proceedings.mlr.press/v54/mcmahan17a/mcmahan17a.pdf](url) ?

**Comment by YuanTingHsieh** [comment, marked as answer]
Please check https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/app_common/aggregators for the aggregation logic on the server side.

https://github.com/NVIDIA/NVFlare/blob/2.3/examples/advanced/cifar10/pt/learners/cifar10_learner.py#L396-L409 this is NOT aggregation, this part of code just load the model from server side
